core/trino-main/src/test/java/io/trino/execution/TestDropColumnTask.java

```diff
@@ -0,0 +1,183 @@
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.metadata.QualifiedObjectName;
+import io.trino.metadata.TableHandle;
+import io.trino.security.AllowAllAccessControl;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.type.RowType;
+import io.trino.spi.type.RowType.Field;
+import io.trino.sql.tree.DropColumn;
+import io.trino.sql.tree.QualifiedName;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.airlift.concurrent.MoreFutures.getFutureValue;
+import static io.trino.spi.StandardErrorCode.COLUMN_NOT_FOUND;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.StandardErrorCode.TABLE_NOT_FOUND;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.testing.TestingHandles.TEST_CATALOG_NAME;
+import static io.trino.testing.assertions.TrinoExceptionAssert.assertTrinoExceptionThrownBy;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Test(singleThreaded = true)
+public class TestDropColumnTask
+        extends BaseDataDefinitionTaskTest
+{
+    @Test
+    public void testDropColumn()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, simpleTable(tableName), false);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).get();
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .isEqualTo(ImmutableList.of(new ColumnMetadata("a", BIGINT), new ColumnMetadata("b", BIGINT)));
+
+        getFutureValue(executeDropColumn(asQualifiedName(tableName), QualifiedName.of("b"), false, false));
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .isEqualTo(ImmutableList.of(new ColumnMetadata("a", BIGINT)));
+    }
+
+    @Test
+    public void testDropOnlyColumn()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, someTable(tableName), false);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).get();
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .isEqualTo(ImmutableList.of(new ColumnMetadata("test", BIGINT)));
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeDropColumn(asQualifiedName(tableName), QualifiedName.of("test"), false, false)))
+                .hasErrorCode(NOT_SUPPORTED)
+                .hasMessageContaining("Cannot drop the only column in a table");
+    }
+
+    @Test
+    public void testDropColumnNotExistingTable()
```

Comment:
`testDropColumnOnMissingTable` maybe - similarly for add columns test also ?

Reply:
"NotExisting" is more common naming in this repository as far as I checked.